test/unit/response.test.js

```diff
@@ -23,6 +24,16 @@ describe('Response', function () {
             expect(jsonified).to.have.property('cookie');
         });
 
+        it('must retain Buffer nature on stream property after new Response(response.toJSON())', function () {
+            var rawResponse = {
+                    body: 'response body',
```

Comment:
The reason I did not want to add an explicit test for the above case, is because we are not really interested in supporting a `{ type: 'Buffer', data: [] }`. What is more important, is to ensure `response` and `new Response(response.toJSON())` are the same, i.e. response instance is properly serialised and deserialised back without loosing data. Which is what the added test ensures.